### PR TITLE
Add LineOfCompressionKernel

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: sumpy/kernel.py [hunk 1]
```diff
@@ -856,6 +856,82 @@ def get_args(self):
                     )
                 ]
 
+
+class ElasticityHelperKernel(ExpressionKernel):
```
Comment (inducer) [inline sumpy/kernel.py:860]: Add to docs? Add docstring to provide context? Test pickling?

File: sumpy/kernel.py [hunk 2]
```diff
@@ -856,6 +857,91 @@ def get_args(self):
...
+    """A helper kernel for the explicit solution to half-space Elasticity problem.
+    This kernel is the line of compression or dilatation of constant strength
+    along the axis "icomp" from zero to negative infinity. See [1] for details.
```
Comment (isuruf) [inline sumpy/kernel.py:864]: I was going by Gimbutas et. al. which doesn't name this kernel, but Mindlin does name this kernel.
Shall I keep the name `ElasticityHelperKernel` or name it something like `LineOfCompressionKernel`?

Comment (inducer) [reply sumpy/kernel.py:864]: I like `LineOfCompression` a lot better than `ElasticityHelper`. :) Include the references for good measure.

File: sumpy/kernel.py [hunk 3]
```diff
@@ -856,6 +857,91 @@ def get_args(self):
...
+        if dim == 3:
+            d = make_sym_vector("d", dim)
+            r = pymbolic_real_norm_2(d)
+            # Kelvin solution
+            expr = d[axis] * var("log")(r + d[axis]) - r
+            scaling = (1 - 2*nu)/(4*var("pi")*mu)
+
+        elif dim is None:
+            expr = None
+            scaling = None
```
Comment (inducer) [inline sumpy/kernel.py:899]: What's this about?

File: test/test_misc.py [hunk 4]
```diff
@@ -398,13 +399,15 @@ def test_elasticity_new():
     stokes_knl = StokesletKernel(3, 0, 1, "mu1", 0.5)
     stokes_knl2 = ElasticityKernel(3, 0, 1, "mu1", 0.5)
     elasticity_knl = ElasticityKernel(3, 0, 1, "mu1", "nu")
+    elasticity_helper_knl = LineOfCompressionKernel(3, 0, "mu1", "nu")
```
Comment (inducer) [inline test/test_misc.py:402]: This should be part of `test_pde_check_kernels`, too. Basic elasticity and Stokes are missing, too, but that's unrelated to this PR. (hence #54)

**Review by inducer — COMMENTED** [review]
Thanks! Two more small wrinkles.

File: sumpy/kernel.py (hunk 3 above)
Comment (isuruf) [reply sumpy/kernel.py:899]: I've copied this from other kernels. I have no idea what's it about

Comment (inducer) [reply sumpy/kernel.py:899]: #55

Comment (isuruf) [reply sumpy/kernel.py]: Thanks. Dropped for the new kernel in 90ddf4f

File: test/test_misc.py (hunk 4 above)
Comment (isuruf) [reply test/test_misc.py:422]: Fixed in 6ef2fad2bc4d91d051eb48145b59f3ff4501ed47

File: test/test_misc.py [hunk 5]
```diff
@@ -33,42 +34,62 @@
...
     def pde_func(self, cp, pot):
-        return cp.laplace(pot) - self.lam**2*pot
+        subs_dict = {sym.Symbol(k): v for k, v in self.extra_kwargs.items()}
+        result = 0
+        for ident, coeff in self.eq.items():
+            lresult = pot
+            for axis, nderivs in enumerate(ident.mi):
+                lresult = cp.diff(axis, lresult, nderivs)
+            result += lresult*float(sym.sympify(coeff).xreplace(subs_dict))
+        return result
```
Comment (inducer) [inline test/test_misc.py:62]: :tada: Nice!

**Review by inducer — APPROVED** [review]
Thanks!

**Comment by inducer** [comment]
It'll go in after #53.

**Pull request merged**